providers/src/airflow/providers/amazon/aws/transfers/sftp_to_s3.py

```diff
@@ -85,6 +89,14 @@ def execute(self, context: Context) -> None:
 
         sftp_client = ssh_hook.get_conn().open_sftp()
 
+        try:
+            sftp_client.stat(self.sftp_path)
+        except FileNotFoundError:
+            if self.fail_on_file_not_exist:
+                raise
+            self.log.info("File %s not found on SFTP server. Skipping transfer.", self.sftp_path)
+            return
```

Comment:
If `self. fail_on_file_not_exist` is `False` then you call `sftp_client.stat(` for nothing. I think the implementation could be simplified and optimized to:



```suggestion
	if self.fail_on_file_not_exist:
		sftp_client.stat(self.sftp_path)
```

Reply:
Your way loses the logging if the file is missing, but it does look much cleaner.

Reply:
@vincbeck 
the purpose of this implementation is to give users an option 'not failing' the dag.

setting 'fail_on_file_not_exist' to 'False' means the user is trying to skip the task and get informed about FileNotFound case, while 'not failing' the dag.

just like @ferruzzi said, the code does look much cleaner but users won't be able to know if the task succeeded because the file existed or just skipped the task because file does not exist.

in my opinion its better to have logging so that users can be informed about the file existence.

Reply:
I personally find it costly to call `sftp_client.stat` just for logging purposes (in case `fail_on_file_not_exist` is `False`). But if you think this is worth it, I am not blocking you :)

Reply:
I also agree with your point. It would be better if paramiko had functionality to check file existence without raising errors, but using sftp_client. stat() was my best.